common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2684,6 +2684,27 @@ def copy_all_asset_metadata(self, source_course_key, dest_course_key, user_id):
                 # update the index entry if appropriate
                 self._update_head(dest_course_key, index_entry, dest_course_key.branch, new_structure['_id'])
 
+    def force_publish_course(self, course_locator, user_id, commit=False):
+        """
+        Helper method to forcefully publish a course,
+        making the published branch point to the same structure as the draft branch.
+        """
+        versions = None
+        index_entry = self.get_course_index(course_locator)
+        if index_entry is not None:
+            versions = index_entry['versions']
+            if commit:
+                # update published branch version only if publish and draft point to different versions
+                if versions['published-branch'] != versions['draft-branch']:
+                    self._update_head(
+                        course_locator,
+                        index_entry,
+                        'published-branch',
+                        index_entry['versions']['draft-branch']
+                    )
+                    return self.get_course_index(course_locator)['versions']
+        return versions
```

Comment:
Should this trigger the `publish` signal? As it stands now, any code (like re-indexing) that triggers during publish won't happen with this command.